### PR TITLE
Make osm namespace generic in docs

### LINKED ISSUE #87: Docs and demo should make osm namespace more generic
**Please describe what should be documented

Currently docs and demos assume osm control-plane is installed in the `osm-system` namespace. We should make this more generic.

**Please suggest where in the repo the document should be located

### PR DESCRIPTION
**Pull request by johnsonshi**
Resolves #87.

Signed-off-by: Johnson Shi <Johnson.Shi@microsoft.com>

### CODE REVIEW

File: content/docs/demos/ingress_contour.md [hunk 1]
```diff
@@ -129,7 +129,7 @@ To proxy connections to TLS backends using HTTPS, the backend service must be an
...
... configuration to use TLS proxying to the backend service, and providing a CA certificate to validate the server certificate presented by the backend service. For this to work, we need to first delegate to Contour the permission to read OSM's CA certificate secret from the OSM's namespace when referenced in the HTTPProxy configuration in the `httpbin` namespace. Refer to the [Upstream TLS section](https://projectcontour.io/docs/v1.18.0/config/upstream-tls/) to learn more about upstream certificate validation and when certificate delegation is necessary. In addition, we must create an IngressBackend resource that specifies HTTPS ingress traffic directed to the `httpbin` service must only accept traffic from a trusted client, osm-contour-envoy in the ingress edge proxy we deployed. OSM automatically provisioned a client certificate for the `osm-contour-envoy` ingress gateway with the Subject Alternative Name (SAN) `osm-contour-envoy.$osm_namespace.cluster.local` during install (where `$osm_namespace` is the OSM namespace env var), so the IngressBackend configuration needs to reference the same SAN for mTLS authentication between the `osm-contour-envoy` edge and the `httpbin` backend.
```
Comment (shashankram) [inline content/docs/demos/ingress_contour.md:132]:
```suggestion
Next, we need to create an HTTPProxy configuration to use TLS proxying to the backend service, and providing a CA certificate to validate the server certificate presented by the backend service. For this to work, we need to first delegate to Contour the permission to read OSM's CA certificate secret from the OSM's namespace when referenced in the HTTPProxy configuration in the `httpbin` namespace. Refer to the [Upstream TLS section](https://projectcontour.io/docs/v1.18.0/config/upstream-tls/) to learn more about upstream certificate validation and when certificate delegation is necessary. In addition, we must create an IngressBackend resource that specifies HTTPS ingress traffic directed to the `httpbin` service must only accept traffic from a trusted client, osm-contour-envoy in the ingress edge proxy we deployed. OSM automatically provisioned a client certificate for the `osm-contour-envoy` ingress gateway with the Subject Alternative Name (SAN) `osm-contour-envoy.<osm-namespace>.cluster.local` during install (where `<osm_namespace>` is the OSM namespace env var), so the IngressBackend configuration needs to reference the same SAN for mTLS authentication between the `osm-contour-envoy` edge and the `httpbin` backend.
```

Similar suggestion elsewhere for denoting the osm namespace outside code blocks in a sentence.

File: content/docs/getting_started/quickstart/manual_demo.md [hunk 2]
```diff
@@ -245,9 +246,9 @@ In permissive traffic policy mode, application connectivity within the mesh is a
   osm install --set=OpenServiceMesh.enablePermissiveTrafficPolicy=true
   ```
 
-1. Post install by patching the `osm-mesh-config` custom resource in the control plane's namespace (`osm-system` by default)
+1. Post install by patching the `osm-mesh-config` custom resource in the control plane's namespace (`osm-system` by default and configurable through $osm_namespace env var)
```
Comment (shashankram) [inline content/docs/getting_started/quickstart/manual_demo.md:249]:
```suggestion
1. Post install by patching the `osm-mesh-config` custom resource in the control plane's namespace (`osm-system` by default and configurable through $osm_namespace environment variable)
```

Same applies to other changes made in this PR.

**Review by shashankram — DISMISSED** [review]

File: content/docs/guides/mesh_config.md [hunk 3]
```diff
@@ -22,22 +28,24 @@ kubectl get meshconfig osm-mesh-config -n osm-system -o yaml
...
 # Validations on the CRD will deny this change
 The MeshConfig "osm-mesh-config" is invalid: spec.traffic.enableEgress: Invalid value: "string": spec.traffic.enableEgress in body must be of type boolean: "string"
 ```
 #### Kubectl Patch Command for Each Key Type
 
+> Note: `<osm-namespace>` refers to the namespace where the osm control plane is installed. By default, the osm namespace is `osm-system`.
+
 | Key | Type | Default Value | Kubectl Patch Command Examples |
 |-----|------|---------------|--------------------------------|
+<<<<<<< HEAD
```
Comment (jaellio) [inline content/docs/guides/mesh_config.md:48]:
```suggestion

```

**Review by shashankram — COMMENTED** [review]
Need to resolve the merge conflict.

File: content/docs/guides/traffic_management/egress.md [hunk 4]
```diff
@@ -50,11 +50,13 @@ Egress can be enabled mesh-wide during OSM install or post install. When egress
...
 
-   `osm-controller` retrieves the egress configuration from the `osm-mesh-config` `MeshConfig` custom resource in its namespace (`osm-system` by default). Use `kubectl patch` to set `enableEgress` to `true` in the `osm-mesh-config` resource.
+   `osm-controller` retrieves the egress configuration from the `osm-mesh-config` `MeshConfig` custom resource in the osm mesh control plane namespace (`osm-system` by default). Use `kubectl patch` to set `enableEgress` to `true` in the `osm-mesh-config` resource.
 
    ```bash
-   kubectl patch meshconfig osm-mesh-config -n osm-system -p '{"spec":{"traffic":{"enableEgress":true}}}' --type=merge
+   kubectl patch meshconfig osm-mesh-config -n <osm-namespace> -p '{"spec":{"traffic":{"enableEgress":true}}}' --type=merge
    ```
+   
+> Note: Replace `<osm-namespace>` with the namespace where osm is installed (by default, osm is installed in `osm-system`).
```
Comment (jaellio) [inline content/docs/guides/traffic_management/egress.md:59]: In other docs we are leaving `osm-system` as is, and adding a comment to `Replace osm-system...`. Does it make sense to be consistent with that here?
```suggestion
> Note: Replace `osm-system` with the namespace where osm is installed (by default, osm is installed in `osm-system`).
```

Comment (shashankram) [reply content/docs/guides/traffic_management/egress.md]: I tried to make this consistent as much as possible. @jaellio let me know if this is still not consistent when used within the code blocks.

File: content/docs/guides/osm_components.md [hunk 5]
```diff
@@ -22,7 +23,7 @@ kubectl get clusterrolebinding,clusterrole,mutatingwebhookconfiguration
 Under the hood, `osm` is using [Helm](https://helm.sh) libraries to create a Helm `release` object in the control plane Namespace. The Helm `release` name is the mesh-name. The `helm` CLI can also be used to inspect Kubernetes manifests installed in more detail. See the Helm docs for how to [install Helm](https://helm.sh/docs/intro/install/).
 
 ```console
-$ helm get manifest osm --namespace osm-system
+$ helm get manifest osm --namespace <osm-namespace>"
```
Comment (jaellio) [inline content/docs/guides/osm_components.md:26]:
```suggestion
$ helm get manifest osm --namespace <osm-namespace>
```

File: content/docs/guides/observability/metrics.md [hunk 6]
```diff
@@ -249,8 +249,9 @@ Ensure that you have followed the steps to run [OSM Demo][2]
 ### Querying proxy metrics for request count
 
 1. Verify that the Prometheus service is running in your cluster
-   - In kubernetes, execute the following command: `kubectl get svc osm-prometheus -n osm-system`
+   - In kubernetes, execute the following command: `kubectl get svc osm-prometheus -n "<osm-namespace>"`.
```
Comment (jaellio) [inline content/docs/guides/observability/metrics.md:252]: Not sure if the quotations around `<osm-namespace>` are intentional.
```suggestion
   - In kubernetes, execute the following command: `kubectl get svc osm-prometheus -n <osm-namespace>`.
```

Comment (shashankram) [reply content/docs/guides/observability/metrics.md]: They aren't, thanks!

**Review by jaellio — APPROVED** [review]

**Pull request merged**